src/createAsyncThunk.test.ts

```diff
@@ -104,3 +109,111 @@ describe('createAsyncThunk', () => {
     expect(errorAction.meta.args).toBe(args)
   })
 })
+
+describe('createAsyncThunk with abortController', () => {
+  const asyncThunk = createAsyncThunk('test', function abortablePayloadCreator(
+    _: any,
+    { signal }
+  ) {
+    return new Promise((resolve, reject) => {
+      if (signal.aborted) {
+        reject(
+          new DOMException(
+            'This should never be reached as it should already be handled.',
+            'AbortError'
+          )
+        )
+      }
+      signal.addEventListener('abort', () => {
+        reject(new DOMException('Was aborted while running', 'AbortError'))
+      })
+      setTimeout(resolve, 100)
+    })
+  })
+
+  let store = configureStore({
+    reducer(store: AnyAction[] = []) {
+      return store
+    }
+  })
+
+  beforeEach(() => {
+    store = configureStore({
+      reducer(store: AnyAction[] = [], action) {
+        return [...store, action]
+      }
+    })
+  })
+
+  test('normal usage', async () => {
+    await store.dispatch(asyncThunk({}))
+    expect(store.getState()).toEqual([
+      expect.any(Object),
+      expect.objectContaining({ type: 'test/pending' }),
+      expect.objectContaining({ type: 'test/fulfilled' })
+    ])
+  })
+
+  test('abort after dispatch', async () => {
+    const thunkAction = asyncThunk({})
+    const promise = store.dispatch(thunkAction)
+    thunkAction.abort('AbortReason')
```

Comment:
The usage here seems unlikely to occur in a real app.  Most code isn't going to reference the `thunkAction` value at all, as it's going to be `dispatch(asyncThunk())`.  

I realize that the abort handling needs to be on a per-call basis, so I'm not sure what the right answer is here.

Reply:
Yeah, I think cancellation is not the most common thing per se, as for many people ignoring earlier running requests might be enough - but if needed this api provides all tools to create a simple reusable abstraction:

```js
function useAbortingApi(asyncApiAction, args) {
  const abortRef = useRef();
  const dispatch = useDispatch();
  return () => {
    const thunkAction = asyncApiAction(args);
    if (abortRef.current) abortRef.current();
    abortRef.current = thunkAction.abort;
    dispatch(thunkAction);
  }
}

function MyComponent(){
  const clickHandler = useAbortingApi(api.fetch, {});
  return <button onClick={clickHandler}>...</button>;
}
```

Reply:
That said, I *could* try to put an `abort` method on the promise returned by the thunk... Might even be pretty simple.

Reply:
That eliminated the "aborted before dispatch" case and allowed for a lot of simplifications.
I hadn't even considered that :man_facepalming:

Reply:
But the diff is ruined now *sigh*